samcli/lib/samlib/resource_metadata_normalizer.py

```diff
@@ -242,29 +242,53 @@ def get_resource_id(resource_properties, logical_id):
             )
             return customer_defined_id
 
+        return logical_id
+
+    @staticmethod
+    def get_resource_name(resource_properties, logical_id):
```

Comment:
if we add this change, the resource name is something related to CDK only. This way will break the Other IaC platforms support. 

The idea is we either define a lambda function using the logical Id or the custom defined Id (IaC Id) using `SamResourceId` metadata. Only in case of CDK, as the `cdk synth` command does not add the `SamSourceId` metadata, we use the `aws:cdk:path` to be the custom defined id (IaC Id).